src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/PathFactory.groovy

```diff
@@ -0,0 +1,35 @@
+package org.openkilda.functionaltests.helpers.model
+
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.PathNodePayload
+import org.openkilda.northbound.dto.v2.flows.FlowPathV2
+import org.openkilda.testing.model.topology.TopologyDefinition
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+class PathFactory {
+    @Autowired
+    TopologyDefinition topologyDefinition
+
+    Path get(List nodes) {
+        if (nodes.isEmpty()) {
+            return new Path([], topologyDefinition)
+        }
+        if (nodes.get(0) instanceof PathNode || nodes.get(0) instanceof FlowPathV2.PathNodeV2) {
+            return new Path(nodes.collect { new FlowPathV2.PathNodeV2(it.getSwitchId(), it.getPortNo(), 0) }, topologyDefinition)
+        }
+        if (nodes.get(0) instanceof PathNodePayload) {
+            def convertedNodes = nodes.collect {
+                [new FlowPathV2.PathNodeV2(it.switchId, it.inputPort == null ? 0 : it.inputPort, 0),
+                 new FlowPathV2.PathNodeV2(it.switchId, it.outputPort == null ? 0 : it.outputPort, 0)]
+            }
+                    .flatten()
```

Comment:
can use `collectMany` instead of `collect` and don't do flatten. Not really an issue as well